substrate-node/pallets/pallet-tfgrid/src/lib.rs

```diff
@@ -2417,41 +2327,24 @@ impl<T: Config> Pallet<T> {
         Ok(name_parsed)
     }
 
-    fn get_public_ip_ip(
-        public_ip_ip: PublicIpIpInput,
-    ) -> Result<PublicIpIpOf<T>, DispatchErrorWithPostInfo> {
-        let public_ip_ip_parsed = <T as Config>::PublicIP::try_from(public_ip_ip)?;
-        Ok(public_ip_ip_parsed)
-    }
-
-    fn get_public_ip_gw(
-        public_ip_gw: PublicIpGatewayInput,
-    ) -> Result<PublicIpGatewayOf<T>, DispatchErrorWithPostInfo> {
-        let public_ip_gw_parsed = <T as Config>::GatewayIP::try_from(public_ip_gw)?;
-        Ok(public_ip_gw_parsed)
-    }
-
     fn get_public_ips(
         public_ips: PublicIpListInput<T>,
-    ) -> Result<PublicIpListOf<T>, DispatchErrorWithPostInfo> {
-        let mut public_ips_list: PublicIpListOf<T> = vec![].try_into().unwrap();
+    ) -> Result<PublicIpListOf, DispatchErrorWithPostInfo> {
+        let mut public_ips_list: PublicIpListOf = vec![].try_into().unwrap();
 
         for ip in public_ips {
-            let public_ip_ip = Self::get_public_ip_ip(ip.ip)?;
-            let public_ip_gateway = Self::get_public_ip_gw(ip.gw)?;
-
             if public_ips_list.contains(&PublicIP {
-                ip: public_ip_ip.clone(),
-                gateway: public_ip_gateway.clone(),
+                ip: ip.ip.clone(),
+                gateway: ip.gw.clone(),
                 contract_id: 0,
             }) {
                 return Err(DispatchErrorWithPostInfo::from(Error::<T>::IpExists));
             }
 
             public_ips_list
                 .try_push(PublicIP {
-                    ip: public_ip_ip,
-                    gateway: public_ip_gateway,
+                    ip: ip.ip,
+                    gateway: ip.gw,
                     contract_id: 0,
                 })
```

Comment:
This block currently creates a stack allocated object, does a memcopy of ip and gateway, takes a reference to the stack allocation for the comparison, removes the stack allocation, and if the ip is not found, creates an exactly new stack allocation with similar content, and moves it to the heap in the vec. This can both be written more concise and slightly more performant by first creating the standalone PublicIP object, and then working with said object:

```
let publicIP = PublicIP {
    ip: ip.ip,
    gateway: ip.gw,
    contract_id: 0,
};

if public_ips_list.contains(&publicIP) { return Err(...) };
public_ips_list.try_push(publicIP).or_else(...);
```